app/components/SignUp/CustomerType.js

```diff
@@ -1,30 +1,80 @@
 import React from 'react';
-import { Field } from 'redux-form';
-import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Field, change, formValueSelector } from 'redux-form';
+import { withTranslation } from 'react-i18next';
+import { CLEAR_ADDRESS_FIELDS, RBU_ADDRESS_FIELDS } from '../../actions/constants/address';
 import Checkbox from '../helix/inputTypes/Checkbox';
 
-const CustomerType = () => {
-  const { t } = useTranslation();
-  return (
-    <div className="customer-type-section">
-      <div className="hxRow">
-        <div className="hxCol hxSpan-4 hxOffset-1">
-          <span className="InputField-label customer-info-header hxRequired">
-            {t('common:account.customer.type')}
-          </span>
-        </div>
-        <div className="hxCol hxSpan-6">
-          <Field
-            name="customerType"
-            content={t('common:account.customer.isRbu')}
-            textField="label"
-            id="customer-type"
-            component={Checkbox}
-          />
+class CustomerType extends React.Component {
+  handleChange = (e) => {
+    if (e.target.checked) {
+      this.populateAddressFields();
+    } else {
+      this.clearAddressFields();
+    }
+  };
+
+  populateAddressFields = () => {
+    Object.keys(RBU_ADDRESS_FIELDS).forEach((field) => {
+      this.props.setAddress(field, RBU_ADDRESS_FIELDS[field]);
+    });
```

Comment:
```suggestion
    Object.entries(RBU_ADDRESS_FIELDS).forEach((entry) => {
      this.props.setAddress(...entry);
    });
```
If you want to get fancy (not a blocker).

Reply:
I like fancy